main/main.cpp

```diff
@@ -1622,35 +1622,21 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 	OS::get_singleton()->set_cmdline(execpath, main_args, user_args);
 
 	{
-		String driver_hints = "";
-#ifdef VULKAN_ENABLED
-		driver_hints = "vulkan";
-#endif
-
-		String default_driver = driver_hints.get_slice(",", 0);
-
-		// For now everything defaults to vulkan when available. This can change in future updates.
-		GLOBAL_DEF("rendering/rendering_device/driver", default_driver);
-		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/rendering_device/driver.windows", PROPERTY_HINT_ENUM, driver_hints), default_driver);
-		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/rendering_device/driver.linuxbsd", PROPERTY_HINT_ENUM, driver_hints), default_driver);
-		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/rendering_device/driver.android", PROPERTY_HINT_ENUM, driver_hints), default_driver);
-		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/rendering_device/driver.ios", PROPERTY_HINT_ENUM, driver_hints), default_driver);
-		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/rendering_device/driver.macos", PROPERTY_HINT_ENUM, driver_hints), default_driver);
-
-		driver_hints = "";
-#ifdef GLES3_ENABLED
-		driver_hints += "opengl3";
-#endif
-
-		default_driver = driver_hints.get_slice(",", 0);
+		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/rendering_device/driver", PROPERTY_HINT_ENUM, "vulkan"), "vulkan");
+		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/rendering_device/driver.windows", PROPERTY_HINT_ENUM, "vulkan"), "vulkan");
+		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/rendering_device/driver.linuxbsd", PROPERTY_HINT_ENUM, "vulkan"), "vulkan");
+		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/rendering_device/driver.android", PROPERTY_HINT_ENUM, "vulkan"), "vulkan");
+		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/rendering_device/driver.ios", PROPERTY_HINT_ENUM, "vulkan"), "vulkan");
+		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/rendering_device/driver.macos", PROPERTY_HINT_ENUM, "vulkan"), "vulkan");
+
+		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver", PROPERTY_HINT_ENUM, "opengl3"), "opengl3");
+		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.windows", PROPERTY_HINT_ENUM, "opengl3,opengl3_angle"), "opengl3");
+		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.linuxbsd", PROPERTY_HINT_ENUM, "opengl3"), "opengl3");
+		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.web", PROPERTY_HINT_ENUM, "opengl3"), "opengl3");
+		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.android", PROPERTY_HINT_ENUM, "opengl3"), "opengl3");
+		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.ios", PROPERTY_HINT_ENUM, "opengl3"), "opengl3");
+		GLOBAL_DEF(PropertyInfo(Variant::STRING, "rendering/gl_compatibility/driver.macos", PROPERTY_HINT_ENUM, "opengl3,opengl3_angle"), "opengl3");
```

Comment:
I guess these should be changed to ``GLOBAL_DEF_RST`` since you will need to restart for them to take effect

Reply:
> How difficult would it be to statically link the libraries? On MacOS, we need to default to ANGLE anyway, so we always have to ship with the option. For Windows, it is pretty important as well.

It should not be hard to do (previous implementation of this PR was doing it), but will create function name conflict and prevent using both native GL and ANGLE in the singe build. DLL should not be an issue, export code is already updated to handle them.

> Reduz has suggested in the chat that he thinks that we should have a fallback on Windows so when OpenGL 3.3 is not supported, we automatically fallback to ANGLE over DX. Do you think doing so will be possible? If not, perhaps we should default to ANGLE on Windows as well

It should be possible (if native GL do not support 3.3 at all), but I have not tested it. For example, on macOS, it will always report 3.2 or 4.1 support, so won't work.